doc/api/cli.md

```diff
@@ -911,6 +911,79 @@ added: v23.6.0
 
 Enable experimental import support for `.node` addons.
 
+### `--experimental-config-file`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.0 - Early development
+
+Use this flag to specify a configuration file that will be loaded and parsed
+before the application starts.
+Node.js will read the configuration file and apply the settings.
+The configuration file should be a JSON file
+with the following structure:
+
+```json
+{
+  "$schema": "https://nodejs.org/dist/REPLACEME/docs/node_config_json_schema.json",
+  "experimental-transform-types": true,
+  "import": [
+    "amaro/transform"
+  ],
+  "disable-warning": "ExperimentalWarning",
+  "watch-path": "src",
+  "watch-preserve-output": true
+}
+```
+
+Only flags that are allowed in [`NODE_OPTIONS`][] are supported.
+No-op flags are not supported.
+Not all V8 flags are currently supported.
+
+It possible to use the [official JSON schema](../node_config_json_schema.json)
+to validate the configuration file, which may vary depending on the Node.js version.
+Each key in the configuration file corresponds to a flag that can be passed
+as a command-line argument. The value of the key is the value that would be
+passed to the flag.
+
+For example, the configuration file above is equivalent to
+the following command-line arguments:
+
+```bash
+node --experimental-transform-types --import amaro/transform
+```
+
+The priority in configuration is as follows:
+
+1. NODE\_OPTIONS and command-line options
+2. Configuration file
+3. Dotenv NODE\_OPTIONS
+
+Values in the configuration file will not override the values in the environment
+variables and command-line options, but will override the values in the `NODE_OPTIONS`
+env file parsed by the `--env-file` flag.
+
+If duplicate keys are present in the configuration file, only
+the first key will be used.
+
+The configuration parser will throw an error if the configuration file contains
+unknown keys or keys that cannot used in `NODE_OPTIONS`.
+
+Node.js will not sanitize or perform validation on the user-provided configuration,
+so **NEVER** use untrusted configuration files.
+In the example below `--some-other-flag`
+will not be sanitized and will be passed to the Node.js process.
+
+```json
+{
+  "import": [
+    "amaro/transform --some-other-flag"
+  ],
+}
+```
```

Comment:
IMO this example shows a bug, not a "feature". `--some-other-flag` is not a valid argument to `--import`.

Reply:
You're basically documenting here that the user can use this as a workaround to pass any flag not allowed at the top level.

Reply:
The env parser will throw an error if the flag is not allowed in node_options. So only options that are already exposed in the config file can be used so there is no real bypass.  We cannot validate the content in the config.json

Reply:
>  We cannot validate the content in the config.json

Is that a temporary limitation or a definitive one?

Reply:
Its not that we can't but it would be a lot of work. Consider that right now we don't for each flag if the string argument is a path, or a known string, or a package name or whatever. Because in the config parser we just know the type of the input (string, number, array) then the validation of the input is performed by the logic of the flag itself

Reply:
You're not really answering my question. I don't mind if we land as status 1.0 without validation, but if we intend to fix that in the future I don't think we should document it so much with an example of how to misuse the feature.

Reply:
My expectations for a config file are much higher than an environment variable. I'm going to block this PR if the goal is not to make it better in the future.

Reply:
Honestly, I'm excited about the feature. But if it stays as it is now, I don't think it should land.

Reply:
~Im not saying we should not improve it~ actually I maybe know how to fix it. Counting the keys parsed and the keys  of the result, if there is a mismatch we throw an error 😬

Reply:
It worked 😄 the problem is fixed, test added and doc updated

Reply:
Wow, thank you!